### ISSUE TITLE
Unable to resolve module `./Libraries/Components/AccessibilityInfo/AccessibilityInfo` from `node_modules\react-native\index.js`

### ISSUE THREAD

**Issue by BartoszKlonowski** [microsoft/react-native-windows#6905]
## Environment
1. `npx react-native --version`:
`4.13.0`
2. `npx react-native info`:
```
System:
    OS: Windows 10 10.0.18363
    CPU: (8) x64 Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
    Memory: 12.94 GB / 23.76 GB
  Binaries:
    Node: 12.9.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.10.2 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.16299.0, 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 16.8.30711.63 (Visual Studio Community 2019)
  Languages:
    Java: Not Found
    Python: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1
    react-native: 0.63.4 => 0.63.4
    react-native-windows: 0.63.18 => 0.63.18
  npmGlobalPackages:
    *react-native*: Not Found
```

- Target Platform Version(s): 10.0.18362
- Target Device(s): PC, Desktop
- Visual Studio Version: 2019
- Build Configuration: Debug

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Create the project for iOS, Android (no Windows yet)
2. Install the packages:
```
{
  "name": "TestsExample",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start"
    "test": "jest"
  },
  "dependencies": {
    "@react-native-community/masked-view": "^0.1.10",
    "@react-navigation/bottom-tabs": "^5.11.1",
    "@react-navigation/drawer": "^5.11.2",
    "@react-navigation/material-bottom-tabs": "^5.3.9",
    "@react-navigation/material-top-tabs": "^5.3.9",
    "@react-navigation/native": "^5.8.9",
    "@react-navigation/stack": "^5.12.6",
    "patch-package": "^6.2.2",
    "postinstall-postinstall": "^2.1.0",
    "react": "16.13.1",
    "react-native": "0.63.4",
    "react-native-appearance": "^0.3.4",
    "react-native-paper": "^4.3.1",
    "react-native-reanimated": "^1.13.1",
    "react-native-redash": "^15.11.1",
    "react-native-safe-area-context": "^3.1.9",
    "react-native-search-bar": "^3.5.1",
    "react-native-vector-icons": "^7.1.0",
    "react-native-webview": "^11.0.0",
    "react-navigation": "^4.4.3",
    "react-navigation-material-bottom-tabs": "^2.3.3",
    "react-navigation-stack": "^2.10.1",
    "react-navigation-tabs": "^2.10.1"
  },
  "devDependencies": {
    "@babel/core": "^7.8.4",
    "@babel/runtime": "^7.8.4",
    "@react-native-community/eslint-config": "^1.1.0",
    "babel-jest": "^25.1.0",
    "babel-plugin-module-resolver": "^4.0.0",
    "babel-preset-expo": "^8.3.0",
    "eslint": "^7.6.0",
    "glob-to-regexp": "^0.4.1",
    "jest": "^25.1.0",
    "metro-react-native-babel-preset": "^0.59.0",
    "react-test-renderer": "16.13.1"
  },
  "jest": {
    "preset": "react-native"
  }
}
```
3. Build and run the application
4. Install Windows:
`npx react-native-windows-init --overwrite --version "0.63.18"`
5. Build and run the application again, but on Windows.

## Expected Results
Application (default app scaffolded when creating the project) should run successfully, or report more detailed error.

## Actual Results
![obraz](https://user-images.githubusercontent.com/70535775/105031864-32267300-5a56-11eb-94c9-583f491d1388.png)

## Troubleshooting
I don't see such an error when creating, compiling and running the application with the same package.json, but on Windows **only** (without previously built for Android/iOS).

**Comment by chrisglein** [comment]
We have a Metro Troubleshooting Guide: http://aka.ms/RNWTroubleshootMetro
Give that a spin and see if it was helpful, report back if you're still blocked.

**Comment by acoates-ms** [comment]
https://microsoft.github.io/react-native-windows/docs/next/metro-config-out-tree-platforms may also help.

**Comment by BartoszKlonowski** [comment]
@chrisglein 
I've ran the metro debugging session and I've ended up with nothing more than just the error message (`Unable to resolve...`).
I've tried to look deeper in call stack for any clue, but there's nothing usefull I could find.

@acoates-ms 
I've checked them all and nothing helps.
The only complication worth mentioning is that  `node -p require.resolve('react-native-windows')`  returns only:
```
[Function: resolve] { paths: [Function: paths] }
```
so I've just hardcoded the path to *node_modules\\react-native-windows*. I hope that's a good approach? The article doesn't specify...

**Comment by acoates-ms** [comment]
Can you provide your metro.config, and your basic repo structure?  Is it a monorepo?  Are you using yarn workspaces etc?

**Comment by BartoszKlonowski** [comment]
@acoates-ms I'm not using Yarn workspaces, and it's not a monorepo.
My project structure looks like this:
```
| DevelopedLibrary/
| --- Example/
| --- --- .vscode/
| --- --- android/
| --- --- ios/
| --- --- src/
| --- --- node_modules/
| --- --- windows/
| --- --- --- .vs/
| --- --- --- AppPackages/
| --- --- --- packages/
| --- --- --- ScreensExample/
| --- --- --- x64
| --- --- --- ScreensExample.sln
| --- --- App.js
| --- --- metro.config.js
| --- --- package.json
| --- --- yarn.lock
| --- --- index.js
| --- --- watchmanconfig.js
| --- --- app.json
```

The *metro.config* is:
```
/**
 * Metro configuration for React Native
 * https://github.com/facebook/react-native
 *
 * @format
 */
const path = require('path');
const blacklist = require('metro-config/src/defaults/blacklist');

module.exports = {
  resolver: {
    blacklistRE: blacklist([
      // This stops "react-native run-windows" from causing the metro server to crash if its already running
      new RegExp(
        `${path.resolve(__dirname, 'windows').replace(/[/\\]/g, '/')}.*`,
      ),
      // This prevents "react-native run-windows" from hitting: EBUSY: resource busy or locked, open msbuild.ProjectImports.zip
      /.*\.ProjectImports\.zip/,
    ]),
    extraNodeModules: {
      'react-native-windows': ".\\node_modules\\react-native-windows"
    }
  },
  watchFolders: [
    // Include hoisted modules
    path.resolve(__dirname, '.\\node_modules\\react-native-windows', 'node_modules'),
  ],
  transformer: {
    getTransformOptions: async () => ({
      transform: {
        experimentalImportSupport: false,
        inlineRequires: false,
      },
    }),
  },
};
```

and the *metro-config/src/defaults/blacklist* is:
```
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 */
"use strict";

var path = require("path"); // Don't forget to everything listed here to `package.json`
// modulePathIgnorePatterns.

var sharedBlacklist = [
  /node_modules\/react\/dist\/.*/,
  /website\/node_modules\/.*/,
  /heapCapture\/bundle\.js/,
  /.*\/__tests__\/.*/
];

function escapeRegExp(pattern) {
  if (Object.prototype.toString.call(pattern) === "[object RegExp]") {
    return pattern.source.replace(/\//g, path.sep);
  } else if (typeof pattern === "string") {
    var escaped = pattern.replace(/[\-\[\]\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&"); // convert the '/' into an escaped local file separator

    return escaped.replace(/\//g, "\\" + path.sep);
  } else {
    throw new Error("Unexpected blacklist pattern: " + pattern);
  }
}

function blacklist(additionalBlacklist) {
  return new RegExp(
    "(" +
      (additionalBlacklist || [])
        .concat(sharedBlacklist)
        .map(escapeRegExp)
        .join("|") +
      ")$"
  );
}

module.exports = blacklist;
```

I hope that's usefull.

---

**NOTE:**
Meanwhile I've noticed the same error, but on a whole different application when trying to debug the [example app](https://github.com/capc0/rn-paper-btn-click-crash-win) of some issue using the steps described in [the wiki](https://github.com/microsoft/react-native-windows/wiki/Contributing-to-the-CLI#example-1-an-app-named-testapp)

What can I do about it? It really slows down my work on many items...
